docs/pages/build-reference/local-builds.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Running builds on your own infrastructure
+---
+
+You can run builds on your local machine or on CI with flag `--local` that will execute the same build process on your infrastructure as our workers
+
+`eas build --platform android --local` or `eas build --platform ios --local`
+
+
+## Prerequisites
+
+- You need to be authenticated (either via regular login or `EXPO_TOKEN`).
+- NPM package `eas-cli-local-build-plugin` needs to be installed and in your `PATH`
+
+## Use cases for local builds
+
+- Privacy concerns or policy of your company on the use of third-party build services. With local builds entire process is happening on your infrastructure and only communication with EAS servers is:
+  - to make sure project `@account/slug` exists
+  - if you are using managed credentials to download them.
+- You don't want to pay for EAS. <!-- and free tier is not enough for you. -->
+- Debugging (more info in [the next section](#using-local-builds-for-debugging))
+
+## Using local builds for debugging
+
+If builds are failing for you on EAS servers it might be easier to debug them locally. To simplify that process we support few environment variables to configure the local build process.
```

Comment:
```suggestion
If builds are failing for you on EAS servers and you're unable to determine the cause from inspecting the logs, it may be helpful to debug the issue locally. To simplify that process we support few environment variables to configure the local build process.
```